ccmlib/cluster.py

```diff
@@ -238,10 +239,22 @@ def add(self, node, is_seed, data_center=None):
             node.set_log_level("TRACE", trace_class)
 
         if data_center is not None:
+            self.debug(f"{node.name}: data_center={node.data_center} rack={node.rack} snitch={self.snitch}")
             self.__update_topology_files()
         node._save()
         return self
 
+    # nodes can be provided in multiple notations, determining the cluster topology:
+    # 1. int - specifying the number of nodes in a single DC, single RACK cluster
+    # 2. list[int] - specifying the number of nodes in a multi DC, single RACK per DC cluster
+    #    The datacenters are automatically named as dc{i}, starting from 1, the rack is named RAC1
+    # 3.a dict[str, int] - specifying the number of nodes in a multi DC, single RACK per DC cluster
+    #     The dictionary keys explicitly identify each datacenter name, and the value is the number of nodes in the DC.
+    # 3.b dict[str, list[int]] - specifying the number of nodes in a multi, multi RACK cluster
+    #     The dictionary keys explicitly identify each datacenter name, and the value is the number of nodes in each RACK in the DC.
+    #     Racks are automatically named as RAC{i}, starting from 1
+    # 3.c dict[str, dict[str, int]] - specifying the number of nodes in a multi DC, multi RACK cluster
+    #     The dictionary keys explicitly identify each datacenter and rack names and the values are the number of nodes in each RACK in the DC.
```

Comment:
would be nice to have examples somewhere as `dict[str, dict[str, int]]` doesn't look friendly

Reply:
added examples in next version